### ISSUE TITLE
Detected usage of a non-validated input variable: $_GET

### ISSUE THREAD

**Issue by fbm-static** [WordPress-Coding-Standards/WordPress-Coding-Standards#392]
The line I have is

```php
<?php $id = intval( $_GET['id'] ); // input var okay ?>
```

But it still has the error.

**Comment by JDGrimes** [comment]
This sniff can be confusing at times because it actually gives two different errors: non-validated, and non-sanitized. In your case, the value is being sanitized (and you shouldn't need the whitelisting comment). But you aren't validating that the `id` GET parameter actually exists before using it:

```php
if ( isset( $_GET['id'] ) ) {
     $id = intval( $_GET['id'] );
}
```

**Comment by fbm-static** [comment]
Ah! Thanks.

**Comment by fbm-static** [comment]
Is doing this necessary when you will only compare it against a string? Like

```
'new' === $_GET['action']
```

Thanks.

**Comment by JDGrimes** [comment]
Validation is still needed, but sanitization is not. See #186, which is fixed in yet-to-be-relased 0.5.0.

**Comment by fbm-static** [comment]
@JDGrimes Thanks for linking into the issue. I appreciate it.

**Comment by sourcerer-mike** [comment]
Reopen this please and accept that the variable is casted to string.

`(string) $_GET['foo']`

**Comment by jrfnl** [comment]
@sourcerer-mike All `$_GET` variables are strings by default when you receive them, so casting them to string makes no difference whatsoever. And you'd still need to check if the array key within `$_GET` is set.

**Comment by Flimm** [comment]
`intval` isn't the only function you can use to sanitise input data. You can also use `sanitize_email`, `sanitize_file_name`, `sanitize_mime_type`, `sanitize_text_field` and a bunch of other functions. For the lurkers coming across this error, read the documentation here:

https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Fixing-errors-for-input-data
